examples/step-74/step-74.cc

```diff
@@ -0,0 +1,1060 @@
+      const unsigned int n_dofs_face    = fe_iv.n_current_interface_dofs();
+      copy_data_face.joint_dof_indices  = fe_iv.get_interface_dof_indices();
+      copy_data_face.cell_matrix.reinit(n_dofs_face, n_dofs_face);
+
+      const std::vector<double> &        JxW     = fe_iv.get_JxW_values();
+      const std::vector<Tensor<1, dim>> &normals = fe_iv.get_normal_vectors();
+
+      const double extent1 =
+        cell->extent_in_direction(GeometryInfo<dim>::unit_normal_direction[f]);
+      const double extent2 = ncell->extent_in_direction(
+        GeometryInfo<dim>::unit_normal_direction[nf]);
+      const double penalty = compute_penalty(fe.get_degree(), extent1, extent2);
+
+      for (unsigned int point = 0; point < n_q_points; ++point)
+        {
+          for (unsigned int i = 0; i < n_dofs_face; ++i)
+            for (unsigned int j = 0; j < n_dofs_face; ++j)
+              copy_data_face.cell_matrix(i, j) +=
+                (-diffusion_coefficient *              // - nu
+                   fe_iv.jump(i, point) *              // [v_h]
+                   (fe_iv.average_gradient(j, point) * // ({grad u_h} .
+                    normals[point])                    //  n)
+
+                 - diffusion_coefficient *               // - nu
+                     (fe_iv.average_gradient(i, point) * // (grad v_h .
+                      normals[point]) *                  //  n)
+                     fe_iv.jump(j, point)                // [u_h]
+
+                 + diffusion_coefficient * penalty * // + nu sigma
+                     fe_iv.jump(i, point) *          // [v_h]
+                     fe_iv.jump(j, point)            // [u_h]
+
+                 ) *
+                JxW[point]; // dx
+        }
+    };
+
+    // The following lambda function will copy data to
+    // the global matrix and right-hand side.
+    // Though there are no hanging node constraints in DG discretization,
+    // we define an empty AffineConstraints oject that
+    // allows us to use copy_local_to_global functionality.
+    AffineConstraints<double> constraints;
+    constraints.close();
+    auto copier = [&](const CopyData &c) {
+      constraints.distribute_local_to_global(c.cell_matrix,
+                                             c.cell_rhs,
+                                             c.local_dof_indices,
+                                             system_matrix,
+                                             system_rhs);
+
+      // Copy data from interior face assembly to the global matrix.
+      for (auto &cdf : c.face_data)
+        {
+          const unsigned int joint_dofs_per_face = cdf.joint_dof_indices.size();
+          for (unsigned int i = 0; i < joint_dofs_per_face; ++i)
+            for (unsigned int k = 0; k < joint_dofs_per_face; ++k)
+              system_matrix.add(cdf.joint_dof_indices[i],
+                                cdf.joint_dof_indices[k],
+                                cdf.cell_matrix(i, k));
+        }
```

Comment:
No, we use the workstream framework. This means the assembly will be done concurrently and each copy task will run as soon as its assembly task is done (but sequentially). If you run with one thread, it will assemble a single cell, then copy, repeat.